Bottom: 2d2e369c79e203c61fa95e4f41a60f1b67974144
Top:    674d7d0df6f31ec4f093a0e78127c11c0de1f2be
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-06-28 15:59:01 +0800

virtio-net: do not reset during xdp set

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index cf6f969..0a30ea9 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1942,6 +1942,7 @@ static int virtnet_xdp_set(struct net_device *dev, struct bpf_prog *prog,
 			return PTR_ERR(prog);
 	}
 
+#if 0
 	/* Changing the headroom in buffers is a disruptive operation because
 	 * existing buffers must be flushed and reallocated. This will happen
 	 * when a xdp program is initially added or xdp is disabled by removing
@@ -1954,6 +1955,7 @@ static int virtnet_xdp_set(struct net_device *dev, struct bpf_prog *prog,
 			goto virtio_reset_err;
 		}
 	}
+#endif
 
 	netif_set_real_num_rx_queues(dev, curr_qp + xdp_qp);
